plugins/reach/admin/CatalogItemImportAction.php

```diff
@@ -0,0 +1,65 @@
+<?php
+/**
+ * @package plugins.reach
+ * @subpackage Admin
+ */
+class CatalogItemImportAction extends KalturaApplicationPlugin
+{
+	/**
+	 * @return string - absolute file path of the phtml template
+	 */
+	public function getTemplatePath()
+	{
+		return realpath(dirname(__FILE__));
+	}
+
+	public function doAction(Zend_Controller_Action $action)
+	{
+		$action->getHelper('layout')->disableLayout();
+		$csvPath = self::getFile();
+
+		$client = Infra_ClientHelper::getClient();
+		$reachPluginClient = Kaltura_Client_Reach_Plugin::get($client);
+		try
+		{
+			$bulkUploadResult = $reachPluginClient->vendorCatalogItem->addFromBulkUpload($csvPath);
+			$bulkUploadId = $bulkUploadResult->id;
+			if ($bulkUploadId)
+			{
+				$action->view->bulkUploadResultId = $bulkUploadId;
+
+				$bulkGetUrl = Infra_ClientHelper::getServiceUrl() . '/api_v3/service/bulkupload_bulk/action/get/id/' . $bulkUploadId . '/ks/' . Infra_ClientHelper::getKs();
+				$action->view->bulkGetUrl = $bulkGetUrl;
+			}
+		}
+		catch (Exception $e)
+		{
+			KalturaLog::err('Error in vendorCatalogItem->addFromBulkUpload ' . $e->getMessage());

```

Comment:
what happens in case of error ? what do u see in the screen?
might want to add something like .... $action->view->errMessage = $e->getMessage();  .... and show it in the admin console.